src/core/tokenHandler.js

```diff
@@ -61,17 +62,17 @@ export default function tokenHandlerFactory(options) {
          *
          * @returns {Promise<String>} the token value
          */
-        getToken: function getToken() {
-            var self = this;
-            var initialToken = options.initialToken;
-
-            var getFirstTokenValue = function getFirstTokenValue() {
-                return tokenStore.dequeue().then(function(currentToken) {
-                    if (currentToken) {
-                        return currentToken.value;
-                    }
-                    return null;
-                });
+        getToken() {
+            const initialToken = options.initialToken;
+
+            const getFirstTokenValue = () => {
```

Comment:
Just a reminder: 
this is a one line function so, you can do an implicit return if you want.

Reply:
indeed, but for readability isn't it better like that? However, we can still get rid of the return while keeping multi-line